### PR TITLE
[docs] Update RFC-0010

### PR DESCRIPTION
**Pull request by delta1**
## Description

Updates the details of the current code organization.

Rendered: https://delta1.github.io/tari/RFC-0010_CodeStructure.html

## Motivation and Context

Doc was slightly outdated.

## How Has This Been Tested?

`cd RFC && mdbook serve`

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I have squashed my commits into a single commit.

### CODE REVIEW

File: RFC/src/SUMMARY.md [hunk 1]
```diff
@@ -1,39 +1,41 @@
...
+  - [RFC-0131: Mining](RFC-0131_Mining.md)
+  - [RFC-0140: Sync and Seeding](RFC-0140_Syncing_and_seeding.md)
+  - [RFC-0150: Wallets](RFC-0150_Wallets.md)
+  - [RFC-0151: Transaction protocol](RFC-0151_TransactionProtocol.md)
+  - [RFC-0152: Emoji ID](RFC-0152_EmojiId.md)
+  - [RFC-0170: Network Communication Protocol](RFC-0170_NetworkCommunicationProtocol.md)
+    - [RFC-0171: Message Serialisation](RFC-0171_MessageSerialisation.md)
+    - [RFC-0172: Peer to Peer Messaging Protocol](RFC-0172_PeerToPeerMessagingProtocol.md)
+  - [RFC-0190: Mempool](RFC-0190_Mempool.md)
+  - [Tari-specific extensions to Mimblewimble](BaseLayerExtensions.md)
```
Comment (stringhandler) [inline RFC/src/SUMMARY.md:21]: Not sure what the correct indentation should here? maybe this?
```suggestion
- [Tari-specific extensions to Mimblewimble](BaseLayerExtensions.md)
```

File: RFC/src/RFC-0010_CodeStructure.md [hunk 2]
```diff
@@ -74,49 +74,73 @@ The domain layer houses the Tari "business logic". All protocol-related concepts
...
+The Tari Protocol code repository is a Rust [workspace] consisting of multiple packages. A package is a set of [crates], which is the source code of a binary or library.
+
+The source code is organized into the following directories.
+
+- `applications` contains crates for all the application-layer executables that form part of the Tari codebase.
+
+  - `tari_base_node` - the Base Node application
+  - `tari_console_wallet` - the CLI Wallet application
+  - `tari_mining_node` - the SHA3 Mining Node (CPU)
+  - `tari_merge_mining_proxy` - the Merge Mining Proxy
```
Comment (stringhandler) [inline RFC/src/RFC-0010_CodeStructure.md:109]: Add `tari_explorer` maybe?

**Review by stringhandler — DISMISSED** [review]

File: RFC/src/SUMMARY.md (hunk 1 above)
Comment (delta1) [reply RFC/src/SUMMARY.md:21]: from [rfc.tari.com](//rfc.tari.com) it looks like it should be in line with mempool (and heads up the 5 extensions) so I think this is right?

image

File: RFC/src/RFC-0010_CodeStructure.md (hunk 2 above)
Comment (delta1) [reply RFC/src/RFC-0010_CodeStructure.md:109]: Sure

**Review by SWvheerden — APPROVED** [review]
lgtm

**Review by philipr-za — APPROVED** [review]
LGTM

**Pull request merged**